### PR TITLE
COMMERCE-4866 Fixed product external reference code update on v1.0

### PR DESCRIPTION
**Pull request by alessiorendina**

### CODE REVIEW

**Comment by Commando18** [comment]
ci:test:sf

**Comment by Commando18** [comment]
ci:test:relevant

**Review by dsitu — APPROVED** [review]

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/ProductResourceImpl.java [hunk 1]
```diff
@@ -819,6 +823,9 @@ private CPDefinition _upsertProduct(Product product) throws Exception {
 	@Reference
 	private CPOptionService _cpOptionService;
 
+	@Reference
+	private CProductLocalService _cProductLocalService;
```
Comment (marco-leo) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/ProductResourceImpl.java:827]: @andrea-ale-sbarra @Commando18 Why LocalService?

**Review by marco-leo — CHANGES_REQUESTED** [review]

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/ProductResourceImpl.java (hunk 1 above)
Comment (Commando18) [reply modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/ProductResourceImpl.java:827]: We don't have a remote service for CProduct. Should we create a method in the CPDefinitionServiceImpl that calls CProductLocalService instead?

Comment (marco-leo) [reply modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/ProductResourceImpl.java:827]: yes please

**Comment by andrea-ale-sbarra** [comment]
snet here https://github.com/liferay-commerce/liferay-portal/pull/510

**Pull request closed without merging**